### ISSUE TITLE
Unhandled exception on QPaintEvent

### ISSUE THREAD

**Issue by DanielMurphy22** [mantidproject/mantid#35029]
This is a hard to reproduce bug, but very possible to stumble across. 

The rough steps are:
- Load data and plot one spectrum. 
- Click the Fit toolbar button to open the FitPropertyBrowser
- Resize the plot window by dragging it to larger/smaller
- Then move the cursor over the plot and off and on the plot
- Eventually a mouse move event will cause the following error 

```

Traceback (most recent call last):
  File "C:\MantidNightlyInstall\bin\lib\site-packages\matplotlib\backends\backend_qt.py", line 272, in enterEvent
    *self.mouseEventCoords(event),
  File "C:\MantidNightlyInstall\bin\lib\site-packages\matplotlib\backends\backend_qt.py", line 265, in mouseEventCoords
    x = pos.x()
AttributeError: 'QPaintEvent' object has no attribute 'x'

Maybe related to this:
https://github.com/matplotlib/matplotlib/issues/11607
```
Only appears to happen on Windows 10, and not in 6.5.0. Possibly could be related to the Pyqt5.15.14 upgrade so interesting to see if the potential downgrade to 5.15.12 will solve this too!

_Originally posted by @thomashampson in https://github.com/mantidproject/mantid/issues/34945#issuecomment-1384260104_

**Comment by DanielMurphy22** [comment]
Found in Core testing and when testing the Mantid Basic Course.

**Comment by MialLewis** [comment]
This bus is easy to reproduce - continuously resize any interface with a `matplotlib` canvas on it.

**Comment by DanielMurphy22** [comment]
Maybe similar to https://github.com/matplotlib/matplotlib/issues/11607

**Comment by MialLewis** [comment]
This is fixed by #35013

**Comment by thomashampson** [comment]
Reopening since this is an issue for moving to qt 5.15.8
